zerver/views/user_settings.py

```diff
@@ -168,14 +168,18 @@ def json_set_avatar(request, user_profile):
         return json_error("You must upload exactly one avatar.")
 
     user_file = list(request.FILES.values())[0]
-    upload_avatar_image(user_file, user_profile, user_profile.email)
-    do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
-    user_avatar_url = avatar_url(user_profile)
-
-    json_result = dict(
-        avatar_url = user_avatar_url
-    )
-    return json_success(json_result)
+    try:
+        upload_avatar_image(user_file, user_profile, user_profile.email)
+        do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+
+        json_result = dict(
+            avatar_url = user_avatar_url
+        )
+        return json_success(json_result)
+    except IOError as e:
```

Comment:
Yeah, this is a pattern we use all over the codebase:
```
tabbott@monastery:~/zulip$ git grep JsonableError | wc
    101     620    8253
```
JsonableError doesn't really have anything directly to do with JSON either -- it perhaps would be better named something like UserError.  Basically it's what we use to cause error message messages to be sent back to the user from any place in the views code paths where an error occurs, regardless of where it is in the views code path.  As one can see in this example, the JsonableError pattern lets us give a specific error message back to the user for a particular error condition, without having to refactor all the functions in between to return a special error message string for this particular case.

(Maybe we should add some documentation on this pattern).